### PR TITLE
Feature: open output PDF with default application

### PR DESCRIPTION
**Pull request by ZaMaZaN4iK**
Hello @manisandro 
Can you review this feature?

### CODE REVIEW

File: qt/src/OutputEditorHOCR.cc [hunk 1]
```diff
@@ -1223,11 +1225,18 @@ void OutputEditorHOCR::savePDF() {
 	if(!failed.isEmpty()) {
 		QMessageBox::warning(m_widget, _("Errors occurred"), _("The following pages could not be rendered:\n%1").arg(failed.join("\n")));
 	}
+
+    bool pdfCanBeOpened = true;
```
Comment (manisandro) [inline qt/src/OutputEditorHOCR.cc]: Can you configure your editor to use tabs for indentation?

File: qt/src/OutputEditorHOCR.cc [hunk 2]
```diff
@@ -1223,11 +1225,18 @@ void OutputEditorHOCR::savePDF() {
...
 	try {
 		document->Close();
 	} catch(PoDoFo::PdfError& e) {
+        pdfCanBeOpened = false;
 		QMessageBox::warning(m_widget, _("Export failed"), _("The PDF export failed (%1).").arg(e.what()));
 	}
+    if(m_pdfExportDialogUi.checkBoxOpenOutputPdf->isChecked() && pdfCanBeOpened) {
+        QDesktopServices::openUrl(QUrl("file:///" + outname));
```
Comment (manisandro) [inline qt/src/OutputEditorHOCR.cc]: You should probably urlencode the outname.

File: qt/src/OutputEditorHOCR.cc (hunk 1 above)
Comment (ZaMaZaN4iK) [reply qt/src/OutputEditorHOCR.cc]: Is it problem?

File: qt/src/OutputEditorHOCR.cc (hunk 2 above)
Comment (ZaMaZaN4iK) [reply qt/src/OutputEditorHOCR.cc]: For what? all works fine now. Do you know better way here?

File: qt/src/OutputEditorHOCR.cc (hunk 1 above)
Comment (manisandro) [reply qt/src/OutputEditorHOCR.cc]: Well, yes, you typically try to not mix indentation stiles within a project, it makes the source files very un-readable.

File: qt/src/OutputEditorHOCR.cc (hunk 2 above)
Comment (manisandro) [reply qt/src/OutputEditorHOCR.cc]: Per RFC 3986 non all characters are valid in a URI and if your filename contains special characters (even just spaces), opening might fail. See QUrl::toPercentEncoding

Comment (ZaMaZaN4iK) [reply qt/src/OutputEditorHOCR.cc]: I found good solution: QDesktopServices::openUrl(QUrl::fromLocalFile(file));

Comment (manisandro) [reply qt/src/OutputEditorHOCR.cc]: Great!

**Comment by manisandro** [comment]
Code looks good, just squash the commits and then feel free to merge!

**Comment by ZaMaZaN4iK** [comment]
Merged. Thank you :-)

**Comment by manisandro** [comment]
I've renamed the commit to more accurately reflect what is done (important since I scan through the commit messages to write the changelog when I do a release).

**Pull request merged**